### PR TITLE
feat(react-textarea): Add shadow variant of filled appearance

### PR DESCRIPTION
**Pull request by sopranopillow**
## Current Behavior

There are no shadows applied to Textarea.

## New Behavior

Textarea now has `filled-shadow` variants.

Note: to see the shadow, you might need to change the background to a lighter color.

image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                |   Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                  Change |
| :----------------------------------------------------------------------------------------------- | -------------------------: | --------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr> | `23.674 kB`<br />`7.83 kB` | `24.407 kB`<br />`7.893 kB` | `733 B` <br />`63 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `188.818 kB`<br />`51.901 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |   `33.19 kB`<br />`10.921 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/9e7682f713135742648c4a6144bce5135048d990'>9e7682f713135742648c4a6144bce5135048d990</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProvider-mount_base.html">2390</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProvider-mount.html">1616</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProvider-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/Avatar-mount_base.html">1319</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/Avatar-mount.html">1302</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/Button-mount_base.html">960</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/Button-mount.html">942</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProvider-mount_base.html">2390</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProvider-mount.html">1616</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProvider-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProviderWithTheme-mount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProviderWithTheme-mount.html">640</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">587</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">586</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">623</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/MakeStyles-mount_base.html">1893</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/MakeStyles-mount.html">1863</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/SpinButton-mount_base.html">2501</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24512/perf-test-react-components/SpinButton-mount.html">2510</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-textarea/src/components/Textarea/Textarea.types.ts
Comment (spmonahan) [inline packages/react-components/react-textarea/src/components/Textarea/Textarea.types.ts:31]: I think this is a fine API change but I also think we should review it as part of a larger team sync since we'll be committing to it for the long haul.

**Review by spmonahan — CHANGES_REQUESTED** [review]
Just blocking this so we can review the API with the larger team.

File: packages/react-components/react-textarea/src/stories/Textarea/TextareaAppearance.stories.tsx
Comment (sopranopillow) [reply packages/react-components/react-textarea/src/stories/Textarea/TextareaAppearance.stories.tsx:60]: yup, I just removed them for the screenshot. The actual example is still the same from #24494.

File: packages/react-components/react-textarea/src/components/Textarea/Textarea.types.ts
Comment (sopranopillow) [reply packages/react-components/react-textarea/src/components/Textarea/Textarea.types.ts:31]: I was also thinking of the prop `shadow`, but it wouldn't make sense since only `filled-darker` and `filled-lighter` have shadows.

Comment (spmonahan) [reply packages/react-components/react-textarea/src/components/Textarea/Textarea.types.ts:31]: Agreed, `shadow` prop is not the way to go. To be clear, I think what you have is where we'll end up but I just want to make sure we get lots of eyes on it first.

File: packages/react-components/react-textarea/src/stories/Textarea/TextareaAppearance.stories.tsx
Comment (sopranopillow) [reply packages/react-components/react-textarea/src/stories/Textarea/TextareaAppearance.stories.tsx:60]: I just realized what I said is wrong 😅 Even if this PR is approved, I'll wait for #24494 to merge.

**Review by micahgodbolt — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Review by spmonahan — APPROVED** [review]

**Pull request merged**
